garglk/sndsdl.c

```diff
@@ -303,17 +428,19 @@ static void music_completion_callback()
 static void sound_completion_callback(int chan)
 {
     channel_t *sound_channel = sound_channels[chan];
-    if (!sound_channel || Mix_Playing(chan))
-    {
-        gli_strict_warning("sound callback failed");
+
+        if (!sound_channel)
+        {
+        fprintf(stderr, "sound_completion_callback called with invalid sound_channel %d\n", chan);
```

Comment:
This ought to be done with `gli_strict_warning`, building up a message first with `snprintf`.  Ideally `gli_strict_warning` would take a format string and variable number of arguments.  I've added #355 to do this.